libpromises/evalfunction.c

```diff
@@ -7651,17 +7650,27 @@ static char *StripPatterns(char *file_buffer, const char *pattern, const char *f
     }
 
     int start, end, count = 0;
+    const size_t original_length = strlen(file_buffer);
     while (StringMatchWithPrecompiledRegex(rx, file_buffer, &start, &end))
     {
-        CloseStringHole(file_buffer, start, end);
+        StringCloseHole(file_buffer, start, end);
 
-        if (count++ > strlen(file_buffer))
+        if (start == end)
         {
+            Log(LOG_LEVEL_WARNING,
+                "Comment regex '%s' matched empty string in '%s'",
+                pattern,
+                filename);
+            break;
+        }
+        assert(start < end);
+        if (count++ > original_length)
+        {
+            debug_abort_if_reached();
             Log(LOG_LEVEL_ERR,
                 "Comment regex '%s' was irreconcilable reading input '%s' probably because it legally matches nothing",
                 pattern, filename);
```

Comment:
Hmm, maybe the log message should come before the abort? The context might be valuable when debugging.

Reply:
Good point. Maybe. The abort gives you a line number, but the strings in the log might be valuable. I usually put these first in the if body, to make it abundantly clear that this is not supposed to happen.

Reply:
I think I will leave it as is. The idea here is that it shouldn't get to this point any longer. If it happens, a developer can move the log message if necessary.